### ISSUE TITLE
[FEATURE REQUEST] Add Support for Shamir Secret Sharing

### ISSUE THREAD

**Issue by DarkKronicle** [Picocrypt/Picocrypt#46]
Thank you so much for your work on this project! 

I was wondering if [Shamir Secret Sharing](https://en.wikipedia.org/wiki/Shamir%27s_secret_sharing) (SSS) could be added. SSS splits a key file into n shares that require k to reconstruct the initial file. 

Currently, it's possible to generate a key file, encrypt with that, and then split up the key file with shamir, but it would be nice to have it all under one umbrella.

---

Below I've put some projects that do something similar and some potential implementations.

- [horcrux](https://github.com/jesseduffield/horcrux) looks like it creates n files, each with the same encrypted data, but has a header at the top including the unique shamir key.
- [RustySecrets](https://github.com/SpinResearch/RustySecrets) adds a signature to hopefully prevent tampering. It doesn't include any of the encrypted data.

Another option would to add some amount of error correction for the header.

**Comment by HACKERALERT** [comment]
Cool but not sure how many people would use :)

**Comment by iPyGuy** [comment]
Yeah, SSS support would be nice but I understand it would not be trivial to add. I remember from my windows days that BestCrypt containers supported SSS and I thought it was so cool, though I’m not sure how it would work for Picocrypt which doesn’t use containers...

**Comment by HACKERALERT** [comment]
A bit out of scope. Let's keep Picocrypt "pico"...
